book/src/developers/designs/replication_coord.md

```diff
@@ -0,0 +1,277 @@
+                               │                  │                                  │
+                               │                  │                                  │
+                            ┌──┘                  │                                  │
+                            │                     │                                  │
+                            │                     │                                  │
+┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┼ ─ ─ ─ ─             │      ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┼ ─ ─ ─ ─
+  replicas_bne              │        │            │        replicas_syd              │        │
+│                           │                     │      │                           │
+    ┌────────┐         ┌────────┐    │            │          ┌────────┐         ┌────────┐    │
+│   │        │         │        │                 │      │   │        │         │        │
+    │   B1   │◀───────▶│   B2   │    │            └──────────│   S1   │◀───────▶│   S2   │    │
+│   │        │         │        │                        │   │        │         │        │
+    └────────┘         └────────┘    │                       └────────┘         └────────┘    │
+│                           ▲                            │        ▲                  ▲
+                            │        │                            │                  │        │
+│                           │                            │        │                  │
+                            ▼        │                            ▼                  ▼        │
+│   ┌────────┐         ┌────────┐                        │   ┌────────┐         ┌────────┐
+    │        │         │        │    │                       │        │         │        │    │
+│   │   B3   │◀───────▶│   B4   │                        │   │   S3   │◀───────▶│   S4   │
+    │        │         │        │    │                       │        │         │        │    │
+│   └────────┘         └────────┘                        │   └────────┘         └────────┘
+                                     │                                                        │
+└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                    └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+```
+
+!!! TBD - How to remove / decomission nodes?
+
+I think origin nodes are persistent and must be manually defined. Will this require configuration of
+their server uuid in the config?
+
+Auto-node groups need to check in with periodic elements, and missed checkins.
+
+Checkins need to send ruv?
+
+If a node misses checkins after a certain period they should be removed from the KRC knowledge?
+
+Should replication maps have "priorities" to make it a tree so that if nodes are offline then it can
+auto-re-route? Should they have multiple paths? Want to avoid loops.
+
+Or is delete of nodes a manual cleanup / triggers clean-ruv?
+
+I think some more thought is needed here. Possibly a node state machine.
```

Comment:
R/O nodes could tombstone after x days of failed checkins, then be auto-reaped after x * 2 or something like that?

I'd say R/W nodes should be manually deleted, it seems weird that they'd have that capability and not require a manual delete step.

Reply:
Yeah I think so too. But I need to really think what that RW delete step will look like too. As well we also need to work out how RO nodes would even work to actually ensure that writes go back to where they need to.

Maybe I'll put these thoughts in the doc. :)